src/extension.ts

```diff
@@ -1531,6 +1531,10 @@ export class ExtensionManager implements vscode.Disposable {
         const testPreset = project.testPreset;
         this.statusBar.setTestPresetName(testPreset?.displayName || testPreset?.name || '');
 
+        if (this.workspaceConfig.automaticReconfigure && presetSelected) {
```

Comment:
I believe extension.ts is not the best place to apply the re-configure. A better option is in cmakeProject.ts where we change the config preset.

Reply:
Do you mean [here](https://github.com/maxmitti/vscode-cmake-tools/blob/auto_configure_after_preset/src/drivers/cmakeDriver.ts#L480-L493) or [here](https://github.com/maxmitti/vscode-cmake-tools/blob/auto_configure_after_preset/src/drivers/cmakeDriver.ts#L495-L510)?

Reply:
I would add that we only want to reconfigure if the preset changed.  It's possible to select the same configurePreset and we would want to do nothing in that case.

We should also reconfigure if the Kit is changed.  You can take a look at the code in `selectKit` to see where we might want to do this.

Reply:
it better be done in `selectConfigurePreset`, where the user is choosing a new preset. 
Also make sure we don't reconfigure if the user is choosing the same preset.

Reply:
Hopefully I chose the right selectConfigurePreset :)